File: apple/PROFI-A/Models/UserProfile.swift
//
//  UserProfile.swift
//  PROFI-A
//
//  Дубликат: ru.profia.app.data.model.UserProfile
//  Для актов: PROFI — только ФИО/тел/email; BUSINESS — + компания + реквизиты.
//

import Foundation

struct UserProfile: Equatable {
    var lastName: String
    var firstName: String
    var middleName: String?
    var email: String
    var phone: String
    var companyName: String?
    var inn: String?
    var kpp: String?
    var legalAddress: String?
    var bankName: String?
    var accountNumber: String?
    var correspondentAccount: String?
    var bic: String?

    /// ФИО одной строкой
    var fullName: String {
        [lastName, firstName, middleName ?? ""]
            .filter { !$0.isEmpty }
            .joined(separator: " ")
    }

    /// Для блока «Исполнитель» в акте: при PROFI — только ФИО; при BUSINESS — companyName или ФИО
    func executorDisplayName(accountType: String) -> String {
        if accountType == "BUSINESS", let name = companyName, !name.isEmpty {
            return name
        }
        return fullName
    }
}

File: app/src/main/java/ru/profia/app/data/model/UserProfile.kt
package ru.profia.app.data.model

data class UserProfile(
    val lastName: String,
    val firstName: String,
    val middleName: String? = null,
    val email: String,
    val phone: String,
    val companyName: String? = null,
    val inn: String? = null,
    val kpp: String? = null,
    val legalAddress: String? = null,
    val bankName: String? = null,
    val accountNumber: String? = null,
    val correspondentAccount: String? = null,
    val bic: String? = null
)

data class AppSettings(
    val city: String = "Краснодар",
    val currency: String = "RUR",
    val language: String = "RU",
    val unitSystem: String = "Метры"
)
